### PR TITLE
Refactor architecture backend

### PR DESCRIPTION
**Pull request by alfredovargasjr**
# Description

POC of refactoring of project.

## Type of change
ASP.NET Core projects (.Net Framework) -> Class Library projects (.Net Framework 6.2)

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

No

# Notes

Proof of concept for layer implementations in project architecture.

# Pivotal Tracker Link:

none

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -34,5 +34,7 @@ msbuild.wrn
 */obj/*
 */*/obj/*
 
+Backend/packages/
```
Comment (julianpoy) [inline .gitignore:37]: Is this necessary? Perhaps I don't understand the purpose of the packages folder.

File: Backend/DataAccessLayer/Class1.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Class1.cs]: What is the purpose of this class? This seems like a placeholder class. Can we make this a placeholder for something we'll need (such as SessionService)?

File: Backend/DataAccessLayer/Data/dbContext.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+using Microsoft.SqlServer.Server;
+
+namespace DataAccessLayer.Data
+{
+    public class DatabaseContext : DbContext
+    {
+        public DbSet<User> Users { get; set; }
+
+
+    }
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Data/dbContext.cs]: This context seems to be missing sessions (it was included in the old context)

File: Backend/DataAccessLayer/Migrations/201811302233048_InitialMigration.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
+namespace DataAccessLayer.Migrations
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Migrations/201811302233048_InitialMigration.cs]: This migration should be named with relation to what it does.

InitialMigration is not descriptive.

Additionally, it seems we lost the migration for the Sessions table (put that table creation statement into a separate migration)

Lastly, can we make the rest of these columns nullable:false?

File: Backend/DataAccessLayer/Migrations/201811302233048_InitialMigration.cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    
+    public partial class InitialMigration : DbMigration
+    {
+        public override void Up()
+        {
+            CreateTable(
+                "dbo.Users",
+                c => new
+                    {
+                        Id = c.Long(nullable: false, identity: true),
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Migrations/201811302233048_InitialMigration.cs]: This shouldn't be a "long". Find a way to enforce type UUIDV4 or similar.

File: Backend/DataAccessLayer/Migrations/201811302233048_InitialMigration.resx [hunk 6]
```diff
@@ -0,0 +1,126 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Migrations/201811302233048_InitialMigration.resx]: I'm pretty sure we're going to want to gitignore this file.

File: Backend/DataAccessLayer/Migrations/201811302311299_ChangeDateTimeToString.cs [hunk 7]
```diff
@@ -0,0 +1,22 @@
+namespace DataAccessLayer.Migrations
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Migrations/201811302311299_ChangeDateTimeToString.cs]: Remove this migration and all associated files before the refactor.

File: Backend/DataAccessLayer/Migrations/Configuration.cs [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    using System;
+    using System.Data.Entity;
+    using System.Data.Entity.Migrations;
+    using System.Linq;
+
+    internal sealed class Configuration : DbMigrationsConfiguration<DataAccessLayer.Data.DatabaseContext>
+    {
+        public Configuration()
+        {
+            AutomaticMigrationsEnabled = false;
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Migrations/Configuration.cs:12]: Why is this false?

File: Backend/DataAccessLayer/Migrations/Configuration.cs [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+        }
+
+        protected override void Seed(DataAccessLayer.Data.DatabaseContext context)
+        {
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Migrations/Configuration.cs:16]: [  NON-BLOCKING ] We should look at getting some seed data in here

File: Backend/ManagerLayer/Class1.cs [hunk 10]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
```
Comment (julianpoy) [inline Backend/ManagerLayer/Class1.cs]: Same comment as earlier. This should be a mock of a more meaningful class, not "Class1"

File: Backend/ManagerLayer/User/Management/UserManagementManager.cs [hunk 11]
```diff
@@ -0,0 +1,34 @@
+﻿using System;
```
Comment (julianpoy) [inline Backend/ManagerLayer/User/Management/UserManagementManager.cs]: This class should be renamed for clarity.

File: Backend/ManagerLayer/User/Management/UserManagementManager.cs [hunk 12]
```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ManagerLayer.User.Management
+{
+    public class UserManagementManager
+    {
+        public void Create()
```
Comment (julianpoy) [inline Backend/ManagerLayer/User/Management/UserManagementManager.cs]: This doesn't look like a manager. This looks like a service.

File: Backend/ManagerLayer/User/UserManager.cs [hunk 13]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ManagerLayer.User
+{
+    public class UserManager
+    {
+        public void Register()
```
Comment (julianpoy) [inline Backend/ManagerLayer/User/UserManager.cs]: [ NON-BLOCKING ] This looks like it should be attached to a router. These methods will not be called by other classes.

File: Backend/ManagerLayer/User/ValidationManager.cs [hunk 14]
```diff
@@ -0,0 +1,16 @@
+﻿using ServiceLayer;
```
Comment (julianpoy) [inline Backend/ManagerLayer/User/ValidationManager.cs]: Shouldn't this be a service as well?

File: Backend/Program/Program.cs [hunk 15]
```diff
@@ -0,0 +1,32 @@
+﻿using DataAccessLayer.Data;
```
Comment (julianpoy) [inline Backend/Program/Program.cs]: What is the purpose of this file?

File: Backend/ServiceLayer/Class1.cs [hunk 16]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
```
Comment (julianpoy) [inline Backend/ServiceLayer/Class1.cs]: Same as above, make this more meaningful or remove it.

File: Backend/ServiceLayer/Services/IPasswordService.cs [hunk 17]
```diff
@@ -6,5 +6,6 @@ public interface IPasswordService
     {
         byte[] GenerateSalt();
         string HashPassword(string password, byte[] salt);
+        object CheckPasswordPwned(string password);
```
Comment (julianpoy) [inline Backend/ServiceLayer/Services/IPasswordService.cs:9]: [ NON-BLOCKING ] The naming of this method could use improvement.

File: Backend/ServiceLayer/Services/PasswordService.cs [hunk 18]
```diff
@@ -1,31 +1,36 @@
...
-                prf: KeyDerivationPrf.HMACSHA1,
-                iterationCount: 10000,
-                numBytesRequested: 256 / 8));
-
-            return hashed;
-        }
-    }
+using System;
+using System.Security.Cryptography;
+ // will need to remove this
```
Comment (julianpoy) [inline Backend/ServiceLayer/Services/PasswordService.cs]: Why?

File: Backend/UnitTesting/UnitTest1.cs [hunk 19]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
```
Comment (julianpoy) [inline Backend/UnitTesting/UnitTest1.cs]: This should be more meaningful. Make this test one of your services or such, even if it's a sanity check (true == true)

**Review by julianpoy — CHANGES_REQUESTED** [review]

**Comment by julianpoy** [comment]
Lastly, this looks like it has conflicts with the base branch. Make sure to merge master into your branch so that it can be merged.

You will have merge conflicts that will need to be resolved...

```
Backend/DataAccessLayer/Configurations/Constants.cs
Backend/DataAccessLayer/Database/DatabaseContext.cs
Backend/DataAccessLayer/Startup.cs
```

File: .gitignore (hunk 1 above)
Comment (alfredovargasjr) [reply .gitignore:37]: that package folder contains builds created from those packages

File: Backend/DataAccessLayer/Class1.cs (hunk 2 above)
Comment (alfredovargasjr) [reply Backend/DataAccessLayer/Class1.cs]: default class created with project, will remove

File: Backend/DataAccessLayer/Migrations/201811302233048_InitialMigration.cs (hunk 5 above)
Comment (alfredovargasjr) [reply Backend/DataAccessLayer/Migrations/201811302233048_InitialMigration.cs]: used Microsofts GUID for sqlserver

File: .gitignore (hunk 1 above)
Comment (julianpoy) [reply .gitignore:37]: :+1:

File: Backend/DataAccessLayer/Class1.cs (hunk 2 above)
Comment (julianpoy) [reply Backend/DataAccessLayer/Class1.cs]: If you wouldn't mind, leave this unresolved until the fix has been pushed. Github should resolve it automatically.

File: Backend/DataAccessLayer/Migrations/201811302233048_InitialMigration.cs (hunk 5 above)
Comment (julianpoy) [reply Backend/DataAccessLayer/Migrations/201811302233048_InitialMigration.cs]: Have you pushed this change? I don't see it.

**Pull request closed without merging**